samtranslator/model/eventsources/pull.py

```diff
@@ -117,6 +117,8 @@ def to_cloudformation(self, **kwargs):
         if self.DestinationConfig:
             if self.DestinationConfig.get("OnFailure") is None:
                 raise InvalidEventException(self.logical_id, "'OnFailure' is a required field for 'DestinationConfig'")
+            if not isinstance(self.DestinationConfig.get("OnFailure"), dict):
+                raise InvalidEventException(self.logical_id, "'OnFailure' has invalid destination configuration")
```

Comment:
Can we state what it should be? If a customer sees this error, do they know how to fix it with this error message?